java-checks/src/main/java/org/sonar/java/checks/NestedSwitchCheck.java

```diff
@@ -19,28 +19,30 @@
  */
 package org.sonar.java.checks;
 
-import java.util.Collections;
+import java.util.Arrays;
 import java.util.List;
 import org.sonar.check.Rule;
 import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
 import org.sonar.plugins.java.api.tree.BaseTreeVisitor;
 import org.sonar.plugins.java.api.tree.ClassTree;
 import org.sonar.plugins.java.api.tree.LambdaExpressionTree;
+import org.sonar.plugins.java.api.tree.SwitchExpressionTree;
 import org.sonar.plugins.java.api.tree.SwitchStatementTree;
+import org.sonar.plugins.java.api.tree.SwitchTree;
 import org.sonar.plugins.java.api.tree.Tree;
 
 @Rule(key = "S1821")
-public class NestedSwitchStatementCheck extends IssuableSubscriptionVisitor {
+public class NestedSwitchCheck extends IssuableSubscriptionVisitor {
 
   @Override
   public List<Tree.Kind> nodesToVisit() {
-    return Collections.singletonList(Tree.Kind.SWITCH_STATEMENT);
+    return Arrays.asList(Tree.Kind.SWITCH_STATEMENT, Tree.Kind.SWITCH_EXPRESSION);
   }
 
   @Override
   public void visitNode(Tree tree) {
     NestedSwitchVisitor visitor = new NestedSwitchVisitor();
-    ((SwitchStatementTree) tree).cases().stream().forEach(c -> c.accept(visitor));
+    ((SwitchTree) tree).cases().stream().forEach(c -> c.accept(visitor));
```

Comment:
I see it was there before. I just don't like when there are more than 2 calls in one line, especially with lambdas. Maybe it will be better if you move `foreach` to the next line.
It minor and just a matter of taste.